tests/test_core.py

```diff
@@ -7383,15 +7391,18 @@ def make_run(fullname, name=-1, compiler=-1, embetter=0, quantum_size=0,
   TT = type(fullname, (T,), dict(run_name = fullname, env = env))
 
   def tearDown(self):
-    super(TT, self).tearDown()
-
-    for k, v in self.env.iteritems():
-      del os.environ[k]
+    try:
+      super(TT, self).tearDown()
+    except Exception, e:
```

Comment:
I did think of that as intentional, however, given that the exception is handled later anyways, I suppose it's better to keep throwing it, so removed the except.